tests/kernel/mem_protect/syscalls/src/main.c

```diff
@@ -42,6 +43,65 @@ static inline size_t z_vrfy_string_nlen(char *src, size_t maxlen, int *err)
 }
 #include <syscalls/string_nlen_mrsh.c>
 
+size_t z_impl_string_nlen_reg(char *src, size_t maxlen, int *err)
+{
+	size_t ret;
+
+	ret = z_user_string_nlen(src, maxlen, err);
+
+/* Part below is made to test that kernel scrubs CPU registers
+ * after returning from the system call
+ */
+#if defined(CONFIG_X86)
+	__asm__ volatile (
+		"movl $0xDEADBEEF, %ebx;\n\t"
```

Comment:
You could use an intermediate to make use of DB_VAL here

Reply:
I can't find out how to use C defined variable in asm code. [DB_VAL] is not working. Do you know how to do that?